kernel/src/actions/visitors.rs

```diff
@@ -544,6 +603,80 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_parse_action_batch_with_sidecar_action() -> DeltaResult<()> {
+        let data = action_batch();
+
+        let mut visitor = SidecarVisitor::default();
+        visitor.visit_rows_of(data.as_ref())?;
+
+        let sidecar1 = Sidecar {
+            path: "016ae953-37a9-438e-8683-9a9a4a79a395.parquet".into(),
+            size_in_bytes: 9268,
+            modification_time: 1714496113961,
+            tags: Some(HashMap::from([(
+                "tag_foo".to_string(),
+                "tag_bar".to_string(),
+            )])),
+        };
+
+        assert_eq!(visitor.sidecars.len(), 1);
+        assert_eq!(visitor.sidecars[0], sidecar1);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_parse_action_batch_with_duplicate_sidecar_actions() -> DeltaResult<()> {
+        let engine = SyncEngine::new();
+        let json_handler = engine.get_json_handler();
+        let json_strings: StringArray = vec![
+            r#"{"sidecar":{"path":"3a0d65cd-4056-49b8-937b-95f9e3ee90e5.parquet","sizeInBytes":9268,"modificationTime":1714496113962,"tags":null}}"#,
+            r#"{"sidecar":{"path":"3a0d65cd-4056-49b8-937b-95f9e3ee90e5.parquet","sizeInBytes":9268,"modificationTime":1714496113962,"tags":null}}"#,
+        ].into();
+
+        let output_schema = get_log_schema().clone();
+        let batch = json_handler
+            .parse_json(string_array_to_engine_data(json_strings), output_schema)
+            .unwrap();
+        let mut visitor = SidecarVisitor::default();
+        visitor.visit_rows_of(batch.as_ref())?;
+
+        let sidecar1 = Sidecar {
+            path: "3a0d65cd-4056-49b8-937b-95f9e3ee90e5.parquet".into(),
+            size_in_bytes: 9268,
+            modification_time: 1714496113962,
+            tags: None,
+        };
+
+        // The second sidecar is a duplicate of the first, so one is ignored
+        assert_eq!(visitor.sidecars.len(), 1);
+        assert_eq!(visitor.sidecars[0], sidecar1);
```

Comment:
Like I mentioned above, I don't think this is a case we want to optimize/test for.